src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.Data.cs

```diff
@@ -17,11 +17,98 @@ namespace Microsoft.NetCore.Analyzers.InteropServices
+                    case RemovedInOSPlatformAttribute:
+                        parsedAttribute.AttributeType = PlatformAttributeType.RemovedInOSPlatformAttribute;
+                        break;
+                    case TargetPlatformAttribute:
+                        parsedAttribute.AttributeType = PlatformAttributeType.TargetPlatformAttribute;
+                        break;
+                    default:
+                        parsedAttribute.AttributeType = PlatformAttributeType.None;
+                        break;
+                }
+
+                if (osAttribute.ConstructorArguments.Length == 1 && osAttribute.ConstructorArguments[0].Type.SpecialType == SpecialType.System_String &&
+                    !osAttribute.ConstructorArguments[0].IsNull && TryParsePlatformNameAndVersion(osAttribute.ConstructorArguments[0].Value.ToString(), out string platformName, out Version? version))
+                {
+                    parsedAttribute.PlatformName = platformName;
+                    parsedAttribute.Version = version;
+                    return true;
+                }
+
+                return false;
+            }
+
+            public override bool Equals(object obj)
+            {
+                if (obj is PlatformAttributeInfo info)
+                {
+                    return Equals(info);
+                }
+                return false;
+            }
+
+            public override int GetHashCode() => HashUtilities.Combine(AttributeType.GetHashCode(), PlatformName.GetHashCode(), Version.GetHashCode());
+
+            public static bool operator ==(PlatformAttributeInfo left, PlatformAttributeInfo right) => left.Equals(right);
+
+            public static bool operator !=(PlatformAttributeInfo left, PlatformAttributeInfo right) => !(left == right);
+
+            public bool Equals(PlatformAttributeInfo other) =>
+                AttributeType == other.AttributeType && IsOSPlatformsEqual(PlatformName, other.PlatformName) && Version.Equals(other.Version);
+        }
+
+        private static bool TryParsePlatformNameAndVersion(string osString, out string osPlatformName, [NotNullWhen(true)] out Version? version)
+        {
+            version = null;
+            osPlatformName = string.Empty;
+            for (int i = 0; i < osString.Length; i++)
+            {
+                if (char.IsDigit(osString[i]))
+                {
+                    if (i > 0 && Version.TryParse(osString.Substring(i), out Version? parsedVersion))
+                    {
+                        osPlatformName = osString.Substring(0, i);
+                        version = parsedVersion;
+                        return true;
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        private struct RuntimeMethodValue : IAbstractAnalysisValue, IEquatable<RuntimeMethodValue>
+        {
```

Comment:
Move to a separate file?

Reply:
Also can be changed to a read only struct?

Reply:
sure it can